Yosemite/Yosemite/SnapshotsProvider/FetchResultSnapshotsProvider.swift

```diff
@@ -189,6 +177,19 @@ public final class FetchResultSnapshotsProvider<MutableType: FetchResultSnapshot
             return nil
         }
     }
+
+    public func updateQuery(_ query: Query) {
+        guard query != self.query else {
+            return
+        }
+        self.query = query
+        fetchedResultsController = createFetchedResultsController(query: query)
+        do {
+            try activateFetchedResultsController()
+        } catch {
+            DDLogError("⛔️ FetchResultSnapshotsProvider: Failed to activate after updating query with error \(error)")
+        }
```

Comment:
⚠️  I'd recommend that we use `restart` instead so that the FRC will not be incorrectly activated. We have to ensure that `start()` was already called beforehand. 

```suggestion
        restartFetchedResultsController()
```

If `start` is not called, we don't want FRC to be active because we will not be observing the `NSManagedObjectContextObjectsDidChange` notifications.

Reply:
Even though the changes for `updateQuery` were reverted in https://github.com/woocommerce/woocommerce-ios/pull/2897/commits/d6f8daf8ff384469d850a857ee718cfc090877ad, we might still need it in the future - e.g. in products tab, we might want to update the query after the user changes the filtering / sorting options when we update it to use diffable table view data source.

Originally, I wanted to call `restartFetchedResultsController` but hesitated on the guard statement `fetchedResultsControllerIsActive` (`fetchedResultsController.fetchedObjects != nil`). What if `activateFetchedResultsController` hasn't been called for the original query for some reason (probably a developer mistake)? What do you think if we call `start()` in case `fetchedResultsControllerIsActive` is `false`? Or there might some undesirable side effects?

```swift
guard fetchedResultsControllerIsActive else {
    start()
    return
}
```